src/main/java/org/spongepowered/common/data/processor/data/entity/SneakingDataProcessor.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data.entity;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spongepowered.common.data.util.DataUtil.getData;
+
+import com.google.common.base.Optional;
+import net.minecraft.entity.Entity;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableSneakingData;
+import org.spongepowered.api.data.manipulator.mutable.entity.SneakingData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.ImmutableDataCachingUtil;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeSneakingData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+
+public class SneakingDataProcessor extends AbstractSpongeDataProcessor<SneakingData, ImmutableSneakingData> {
+
+    @Override
+    public Optional<SneakingData> createFrom(DataHolder dataHolder) {
+        if (dataHolder instanceof Entity) {
+            final boolean sneaking = ((Entity) dataHolder).isSneaking();
+            return Optional.<SneakingData>of(new SpongeSneakingData(sneaking));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof Entity;
+    }
+
+    @Override
+    public Optional<SneakingData> from(DataHolder dataHolder) {
+        if (dataHolder instanceof Entity) {
+            final boolean sneaking = ((Entity) dataHolder).isSneaking();
+            return Optional.<SneakingData>of(new SpongeSneakingData(sneaking));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<SneakingData> fill(DataHolder dataHolder, SneakingData manipulator, MergeFunction overlap) {
+        if (dataHolder instanceof Entity) {
+            final Optional<SneakingData> oldData = from(dataHolder);
+            final SneakingData newData = checkNotNull(overlap, "Merge function was null!").merge(oldData.orNull(), manipulator);
```

Comment:
The merge is flipped. You're trying to fill the provided `SneakingData` with data from the `Entity`.